reservation_units/querysets.py

```diff
@@ -0,0 +1,418 @@
+from copy import copy
+from datetime import date, datetime, time, timedelta
+from decimal import Decimal
+from math import ceil
+from typing import TYPE_CHECKING, Self, TypedDict
+
+from django.db import models
+from django.db.models import Case, Prefetch, Q, Value, When
+from django.utils import timezone
+from django.utils.timezone import get_default_timezone
+from elasticsearch_django.models import SearchResultsQuerySet
+
+from applications.choices import ApplicationRoundStatusChoice
+from applications.models import ApplicationRound
+from opening_hours.models import ReservableTimeSpan
+from opening_hours.utils.reservable_time_span_client import TimeSpanElement, override_reservable_with_closed_time_spans
+
+if TYPE_CHECKING:
+    from opening_hours.models import ReservationUnit
+
+DEFAULT_TIMEZONE = get_default_timezone()
+
+INTERVAL_TO_MINUTES: dict[str, int] = {
+    "interval_15_mins": 15,
+    "interval_30_mins": 30,
+    "interval_60_mins": 60,
+    "interval_90_mins": 90,
+}
+
+
+def _get_hard_closed_time_spans_for_reservation_unit(reservation_unit: "ReservationUnit") -> list[TimeSpanElement]:
+    reservation_unit_closed_time_spans: list[TimeSpanElement] = []
+
```

Comment:
In the situation that `reservation_ends < reservation_begins` (reservation period will/has recommence/d) need to add just one hard closed time span based on those variables (instead of the two semi-open ranges):
```
TimeSpanElement(
	start_datetime=reservation_unit.reservation_ends,
	end_datetime=reservation_unit.reservation_begins,
	is_reservable=False,
)
```

(The edge case -- incorrectly set values really -- of `reservation_ends = reservation_begins` would be correctly handled already, by being treated as always closed: datetime.min -> begins/ends -> datetime.max.)